anchore_engine/services/catalog/__init__.py

```diff
@@ -974,6 +982,12 @@ def handle_image_watcher(*args, **kwargs):
                     )
                     raise
 
+                if not catalog_impl.is_image_valid_size(image_info):
+                    localconfig = anchore_engine.configuration.localconfig.get_config()
+                    raise Exception(
```

Comment:
consider raising an AnchoreApiError here, might be a BadRequest?

Reply:
Yeah I was actually curious about this one. I initially raised an api exception, but then changed it because this exception is getting raised within a loop of all the detected tags. Its set up in such a way that all errors in this block will be caught and the process will then continue on to the next tag so this exception never gets exposed through the API.

Reply:
Generally we want to keep API-specific errors out of code that isn't in the API tier. If invoked via an API then an internal Exception can be mapped to an API exception if that error should be exposed to users.